R/compat-dplyr.R

```diff
@@ -0,0 +1,39 @@
+dribble_maybe_reconstruct <- function(data, template) {
+  if (dribble_is_reconstructable(data)) {
+    # in workflowsets, davis uses new_workflow_set0(x) here
+    new_dribble(data)
+  } else {
+    # davis says:
+    # in dribble_maybe_reconstruct(), rather than as_tibble() you may want to
+    # use new_tibble(). I think there is a chance that the input could actually
+    # just be a list, rather than a data frame, so really you are just adding
+    # the appropriate data frame specific attributes back on the list.
```

Comment:
@DavisVaughan Are you saying that some internal machinations inside dplyr or vctrs could lead to an input that's a list? And, if that's the case, can I rely on the list components having the same length? I see that you check for a list in workflowsets, but you don't check for same length.

Reply:
Yea, exactly that. I can't remember exactly when it could occur, but, if it did happen, the unclassed list would most likely come through as `x` in your `vec_restore()` method.

The components should all still be guaranteed to have the same length, so checking that shouldn't be required.